lib/vagrant-libvirt/driver.rb

```diff
@@ -135,7 +135,9 @@ def state(machine)
         end
 
         # TODO: terminated no longer appears to be a valid fog state, remove?
-        return :not_created if domain.nil? || domain.state.to_sym == :terminated
+        return :not_created if domain.nil?
+        return nil if domain.state.nil?
```

Comment:
While the provider code correctly handles nil, I do think it might be clearer to return `:unknown` here rather than rely on the subsequent conversion

Reply:
Looks like the unit tests need some adjustments as the expected state being passed in no longer matches in a few cases resulting in slightly different behaviour. Possibly also the tests are a little fragile in a couple of cases

Reply:
Thank you for feedback.
* I have translated the nil value to `:unknown` in the driver as requested.
* Added a unit test for nil handling.
* Made some unit tests less brittle with respect to domain.state call count.